From 3cd403fd0979328f9c6f711a3983c64c250ce7df Mon Sep 17 00:00:00 2001
From: Thijs Schreijer <thijs@thijsschreijer.nl>
Date: Thu, 25 Jun 2015 00:14:50 +0200
Subject: [PATCH 4/4] config.lua is not being integrated, bug introduced in
 #385

(Upstream patch taken from master branch)
(cherry picked from commit f6974b40735b9300f7b529da6e602459d0db9c49)
---
 src/luarocks/cfg.lua | 19 ++++++++++++-------
 1 file changed, 12 insertions(+), 7 deletions(-)
 mode change 100644 => 100755 src/luarocks/cfg.lua

diff --git a/src/luarocks/cfg.lua b/src/luarocks/cfg.lua
old mode 100644
new mode 100755
index 5dcd59c..3cf0565
--- a/src/luarocks/cfg.lua
+++ b/src/luarocks/cfg.lua
@@ -159,6 +159,16 @@ env_for_config_file = {
      print(util.show_table(env_for_config_file, "global environment"))
    end,
 }
+-- Merge values from config files read into the `cfg` table
+local merge_overrides = function(overrides)
+   if overrides.rocks_trees then
+      cfg.rocks_trees = nil
+   end
+   if overrides.rocks_servers then
+      cfg.rocks_servers = nil
+   end
+   util.deep_merge(cfg, overrides)
+end
 
 sys_config_file = site_config.LUAROCKS_SYSCONFIG or sys_config_dir.."/config-"..cfg.lua_version..".lua"
 local err, errcode
@@ -172,6 +182,7 @@ if (not sys_config_ok) and errcode ~= "open" then -- either "load" or "run"; bad
    io.stderr:write(err.."\n")
    os.exit(cfg.errorcodes.CONFIGFILE)
 end
+   merge_overrides(sys_config_ok)
 
 if not site_config.LUAROCKS_FORCE_CONFIG then
 
@@ -189,13 +200,7 @@ if not site_config.LUAROCKS_FORCE_CONFIG then
    end
    if home_overrides then
       home_config_ok = true
-      if home_overrides.rocks_trees then
-         cfg.rocks_trees = nil
-      end
-      if home_overrides.rocks_servers then
-         cfg.rocks_servers = nil
-      end
-      util.deep_merge(cfg, home_overrides)
+      merge_overrides(home_overrides)
    else
       home_config_ok = home_overrides
       if errcode ~= "open" then
-- 
2.4.4

